File: jput-core/src/main/java/cz/encircled/jput/JPutCommons.java
package cz.encircled.jput;

/**
 * @author Vlad on 27-May-17.
 */
public class JPutCommons {

    public static void validatePercentile(double percentile) {
        if (percentile <= 0D || percentile > 1D) {
            throw new IllegalStateException("Wrong percentile [" + percentile + "], must be [0 < Q <= 1]");
        }
    }

}

File: jput-core/src/main/java/cz/encircled/jput/JPutCommons.kt
package cz.encircled.jput

/**
 * @author Vlad on 27-May-17.
 */
object JPutCommons {

    fun validatePercentile(rank: Double) {
        check(!(rank <= 0.0 || rank > 1.0)) { "Wrong percentile [$rank], must be [0 < Q <= 1]" }
    }

}
